pkg/metapage/slot.go

```diff
@@ -87,11 +87,11 @@ func (m *LinkedMetaSlot) Reset() error {
 	return m.pager.Reset(m.offset)
 }
 
-// Collect returns a slice of all linked meta pages from this page to the end.
+// Collect returns a slice of all linked meta slots from this page to the end.
 // This function is useful for debugging and testing, however generally it should
 // not be used for functional code.
 func (m *LinkedMetaSlot) Collect() ([]*LinkedMetaSlot, error) {
-	var pages []*LinkedMetaSlot
+	var slots []*LinkedMetaSlot
 	node := m
 	for {
 		exists, err := node.Exists()
```

Comment:
so it's correctly adding the slots in `AddNext()`, however when we `Collect()`, it's an infinite loop because this isn't catching.